core/trino-main/src/test/java/io/trino/operator/output/BenchmarkPartitionedOutputOperator.java

```diff
@@ -43,110 +47,261 @@
+import static io.trino.spi.type.SmallintType.SMALLINT;
+import static io.trino.spi.type.TestingTypeManager.createMapType;
 import static io.trino.spi.type.VarcharType.VARCHAR;
-import static java.nio.charset.StandardCharsets.ISO_8859_1;
+import static io.trino.testing.TestingSession.testSessionBuilder;
+import static java.util.Collections.nCopies;
+import static java.util.Collections.unmodifiableList;
 import static java.util.concurrent.Executors.newCachedThreadPool;
 import static java.util.concurrent.Executors.newScheduledThreadPool;
 import static java.util.concurrent.TimeUnit.MILLISECONDS;
 
 @State(Scope.Thread)
 @OutputTimeUnit(MILLISECONDS)
-@Fork(2)
-@Warmup(iterations = 20, time = 500, timeUnit = MILLISECONDS)
-@Measurement(iterations = 20, time = 500, timeUnit = MILLISECONDS)
+@Fork(3)
+@Warmup(iterations = 5, time = 500, timeUnit = MILLISECONDS)
+@Measurement(iterations = 10, time = 500, timeUnit = MILLISECONDS)
 @BenchmarkMode(Mode.AverageTime)
 public class BenchmarkPartitionedOutputOperator
 {
     private static final OperatorFactories OPERATOR_FACTORIES = new TrinoOperatorFactories();
 
-    private final OperatorFactories operatorFactories;
-
-    public BenchmarkPartitionedOutputOperator()
-    {
-        this(OPERATOR_FACTORIES);
-    }
-
-    protected BenchmarkPartitionedOutputOperator(OperatorFactories operatorFactories)
+    @Benchmark
+    public void addPage(BenchmarkData data)
     {
-        this.operatorFactories = operatorFactories;
+        addPage(data, OPERATOR_FACTORIES, testSessionBuilder());
     }
 
-    @Benchmark
-    public void addPage(BenchmarkData data)
+    protected void addPage(BenchmarkData data, OperatorFactories operatorFactories, SessionBuilder sessionBuilder)
     {
-        PartitionedOutputOperator operator = data.createPartitionedOutputOperator(operatorFactories);
+        PartitionedOutputOperator operator = data.createPartitionedOutputOperator(operatorFactories, sessionBuilder);
         for (int i = 0; i < data.getPageCount(); i++) {
             operator.addInput(data.getDataPage());
         }
         operator.finish();
     }
 
+    @Test
+    public void verifyAddPage()
+    {
+        BenchmarkData data = new BenchmarkData();
+        data.setup();
+        new BenchmarkPartitionedOutputOperator().addPage(data);
+    }
+
     @State(Scope.Thread)
+    @SuppressWarnings("unused")
     public static class BenchmarkData
     {
-        private static final int PAGE_COUNT = 5000;
-        private static final int PARTITION_COUNT = 512;
-        private static final int ENTRIES_PER_PAGE = 256;
-        private static final DataSize MAX_MEMORY = DataSize.of(1, GIGABYTE);
-        private static final RowType rowType = RowType.anonymous(ImmutableList.of(VARCHAR, VARCHAR, VARCHAR, VARCHAR));
-        private static final List<Type> TYPES = ImmutableList.of(BIGINT, rowType, rowType, rowType);
+        private static final int PARTITION_COUNT = 256;
```

Comment:
We should parametrize partition count as well

Reply:
No one runs all those benchmarks at once. I have benchmarks that takes weeks to run. You can select params based on your current needs.
I guess the numbers that I would go with are 1, 16, 256 just to see the difference